php/libraries/NDB_BVL_Feedback.class.inc

```diff
@@ -941,91 +941,81 @@ class NDB_BVL_Feedback
      *
      * @param integer $feedbackID The ID of the feedback thread being closed
      *
-     * @return void
-     *  FIXME: This shares a lot of code from activateThread, it should be
-     *         refactored
-     * FIXME: This needs to properly check for sites.
+     * @return int The number of updated threads
      */
-    function closeThread($feedbackID=null)
+    function closeThread(int $feedbackID = null)
     {
         // new DB Object
         $db =& Database::singleton();
 
-        $setArray = array(
-            "Status" => 'closed',
-            "UserID" => $this->_username,
-        );
-
-        $whereArray = array(
-            "Public" => 'Y',
-            "Active" => 'Y',
-        );
+        $whereStr = 'Public = \'Y\' AND Active = \'Y\' ';
         if (!empty($feedbackID)) {
             // only one thread to be closed
-            $whereArray = array_merge(
-                $whereArray,
-                array("FeedbackID" => $feedbackID)
-            );
+            $whereStr .= 'AND FeedbackID = :v_feedbackid ';
+            $params    = array('v_feedbackid' => $feedbackID);
```

Comment:
What happens if `$feedbackID` is null?

Reply:
I think it is meant to close more thant one thread.  i.e.: All threads related to either the commentid, the sessionid of the candid.

Reply:
It might be good to note that in the function description. As you said, it's quite powerful.

Reply:
But I can't find any place where it is called without the feedbackid.

Reply:
Maybe it should just be `int` then with no default param? Might as well eliminate that potential for bugs if it's not used anyway.

Reply:
It won't be done in this PR.